core/DataTable/Filter/Pattern.php

```diff
@@ -93,4 +96,30 @@ public function filter($table)
             }
         }
     }
+
+    /**
+     * See {@link Pattern}.
+     *
+     * @param array $array
+     * @return array
+     */
+    public function filterArray($array)
```

Comment:
The first solution that I implemented used this `filter_pattern` and `filter_column[]`. For a more efficient solution we do now apply the `pattern` in the API method itself. In theory this here is no longer needed but thought it would be still useful to have and left it here therefore. Also wrote a couple of tests for it.